lib/debts/letter_downloader.rb

```diff
@@ -38,5 +40,13 @@ def list_letters
         )
       end
     end
+
+    private
+
+    def verify_letter_in_folder(document_id)
+      raise Common::Exceptions::Unauthorized unless list_letters.map do |letter|
```

Comment:
using `map` and `include?` is iterating over this list more than once.  `any` accepts a block and still returns a bool and IMO is a little clearer
```
raise Common::Exceptions::Unauthorized unless list_letters.any? { |letter| letter[:document_id] == document_id }
```